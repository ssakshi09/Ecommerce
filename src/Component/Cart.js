import React from "react";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper,
  Button,
} from "@material-ui/core";
import { combineitems } from "./combineitems";

function Cart({ items, handleRemoveFromCart }) {
  const combinedItems = combineitems(items);
  return (
    <Paper>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Product</TableCell>
            <TableCell>Price</TableCell>
            <TableCell>Quantity</TableCell>
            <TableCell>Total</TableCell>
            <TableCell>Action</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {combinedItems.map((item) => (
            <TableRow key={item.id}>
              <TableCell>{item.title}</TableCell>
              <TableCell>${item.price}</TableCell>
              <TableCell>{item.quantity}</TableCell>
              <TableCell>${item.total}</TableCell>
              <TableCell>
                <Button onClick={() => handleRemoveFromCart(item)}>
                  Remove
                </Button>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  );
}

export default Cart;
