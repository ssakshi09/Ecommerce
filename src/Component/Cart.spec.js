import Cart from "./Cart";
import { render, fireEvent, screen, shallow } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

const handleRemoveFromCart = jest.fn();

describe("Cart Component", () => {
  const items = [
    {
      id: 1,
      title: "iphone",
      description: "An apple mobile which is not sing like apple",
      price: "$549",
      quantity: 1,
      total: "$549",
    },
  ];
  it("Should Match Snapshot", () => {
    const wrapper = render(
      <Cart items={items} handleRemoveFromCart={handleRemoveFromCart} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should call the handleRemoveFromCart function when the remove button is clicked", () => {
    const items = [{ id: 1, title: "Item 1", price: 10, quantity: 1 }];
    const { getByText } = render(
      <Cart items={items} handleRemoveFromCart={handleRemoveFromCart} />
    );
    fireEvent.click(getByText("Remove"));
    expect(handleRemoveFromCart).toHaveBeenCalledWith({
      id: 1,
      title: "Item 1",
      price: 10,
      quantity: 1,
      total: 10,
    });
  });
});
