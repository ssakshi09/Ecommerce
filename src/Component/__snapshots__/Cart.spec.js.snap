// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component Should Match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Product
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Price
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Quantity
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Total
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Action
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                iphone
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                $
                $549
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                $
                NaN
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Remove
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Product
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Price
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Quantity
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Total
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              Action
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              iphone
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              $
              $549
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              $
              NaN
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Remove
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
